### PR TITLE
VPC: Add Security Group resources and constants/enums

### PR DESCRIPTION
**Pull request by cjschaef**
Add resource definitions for managing VPC Security Groups and Rules, along with a set of enumerations to simplify validation and options.

**What this PR does / why we need it**:  Adds support to define SecurityGroups in VPC for VPCCluster definitions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # N/A

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: api/v1beta2/types.go [hunk 1]
```diff
@@ -140,12 +142,186 @@ var (
...
+
+// SecurityGroupRuleProtocol represents the protocols for a Security Group Rule.
+// +kubebuilder:validation:Enum=all;icmp;tcp;udp
+type SecurityGroupRuleProtocol string
+
+const (
+	// SecurityGroupRuleProtocolAll defines the Rule is for all network protocols.
+	SecurityGroupRuleProtocolAll SecurityGroupRuleProtocol = vpcv1.NetworkACLRuleProtocolAllConst
+	// SecurityGroupRuleProtocolICMP defiens the Rule is for ICMP network protocol.
+	SecurityGroupRuleProtocolICMP SecurityGroupRuleProtocol = vpcv1.NetworkACLRuleProtocolIcmpConst
```
Comment (Prajyot-Parab) [inline api/v1beta2/types.go:175]:
```suggestion
	SecurityGroupRuleProtocolICMP SecurityGroupRuleProtocol = vpcv1.NetworkACLRuleProtocolICMPConst
```
lets maintain uniformity

**Review by Prajyot-Parab — CHANGES_REQUESTED** [review]

File: api/v1beta2/types.go (hunk 1 above)
Comment (cjschaef) [reply api/v1beta2/types.go:175]: That is the name of the constant in the VPC code, I cannot change that
```
// Constants associated with the NetworkACLRule.Protocol property.
// The protocol to enforce.
const (
	NetworkACLRuleProtocolAllConst  = "all"
	NetworkACLRuleProtocolIcmpConst = "icmp"
	NetworkACLRuleProtocolTCPConst  = "tcp"
	NetworkACLRuleProtocolUDPConst  = "udp"
)
```

Comment (Prajyot-Parab) [reply api/v1beta2/types.go:175]: If it is referred as `Icmp` in that package, lets do change in our code to follow same standard wherever possible.

Comment (cjschaef) [reply api/v1beta2/types.go:175]: I think the standard is `ICMP` actually, which is also what the docs follow too. I think the SDK itself is inaccurate.
image

**Comment by Prajyot-Parab** [comment]
@Karthik-K-N @Amulyam24 ptal

File: api/v1beta2/types.go (hunk 1 above)
Comment (cjschaef) [reply api/v1beta2/types.go]: I changed this one reference to `Icmp` to match the SDK constant name, have left remaining references as `ICMP`

File: api/v1beta2/types.go [hunk 2]
```diff
@@ -142,12 +144,186 @@ var (
...
+	MinimumPort int `json:"minimumPort,omitempty"`
+}
+
+// SecurityGroup defines a VPC Security Group that should exist or be created within the specified VPC, with the specified Security Group Rules.
+type SecurityGroup struct {
+	// id of the Security Group.
+	ID *string `json:"id,omitempty"`
+
+	// name of the Security Group.
+	Name *string `json:"name,omitempty"`
```
Comment (Karthik-K-N) [inline api/v1beta2/types.go:231]: Dont we need +optional tags here?

**Review by Karthik-K-N — COMMENTED** [review]
A small doubt, otherwise overall LGTM

File: api/v1beta2/types.go (hunk 2 above)
Comment (cjschaef) [reply api/v1beta2/types.go:231]: Yes, let me add those tags quick.

Comment (cjschaef) [reply api/v1beta2/types.go:234]: updated

**Review by Karthik-K-N — COMMENTED** [review]
/lgtm

**Review by Prajyot-Parab — APPROVED** [review]

**Pull request merged**